app/change_set_persisters/plum_change_set_persister/apply_ark_by_class.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+class PlumChangeSetPersister
+  class ApplyArkByClass
+    attr_reader :change_set_persister, :change_set
+    def initialize(change_set_persister:, change_set:, post_save_resource: nil)
+      @change_set = change_set
+      @change_set_persister = change_set_persister
+    end
+
+    def run
+      return unless supported_classes.include?(change_set.resource.class) && !(supported_states & change_set.resource.state).empty?
+      change_set.validate(identifier: updated_identifiers)
+      change_set.sync
+      change_set
+    end
+
+    private
+
+      def supported_classes
+        [SimpleResource]
+      end
+
+      def supported_states
```

Comment:
I've been working on putting information about states into the workflow itself so that adding a new workflow doesn't have to mean sprinkling different kinds of logic around the codebase. This might be a good thing to fit into what i'm doing with #1065